src/smt/assertions.cpp

```diff
@@ -59,8 +61,8 @@ void Assertions::finishInit()
   {
     // In the case of incremental solving, we appear to need these to
     // ensure the relevant Nodes remain live.
-    d_assertionList = new (true) AssertionList(d_userContext);
-    d_globalDefineFunRecLemmas.reset(new std::vector<Node>());
+    d_assertionList = new (true) AssertionList(d_env.getUserContext());
```

Comment:
What exactly does `new (true) ...` do and why do we use it here?

Reply:
I believe this has something to do with dynamically allocating a context-dependent object.

I'm not sure why we have this strange way of doing this, I can just update to a more normal way of doing this.